tests/macros/t18235.nim

```diff
@@ -1,10 +1,18 @@
-discard """
-errormsg: '''
-undeclared identifier: 'foo'
-'''
-"""
 import m18235
 
 # this must error out because it was never actually exported
+doAssert(not declared(foo))
```

Comment:
optional nit, but `doAssert not declared(foo)` is the more common style in tests

Reply:
it cant hurt to test some other style too